qdev_wrappers/station_configurator.py

```diff
@@ -0,0 +1,161 @@
+from .parameters import DelegateParameter
+
+log = logging.getLogger(__name__)
+
+# config from the qcodesrc.json file (working directory, home or qcodes dir)
+enable_forced_reconnect = qcodes.config["user"]["enable_forced_reconnect"]
+
+
+class StationConfigurator:
+    """
+    The StationConfig class enables the easy creation of intstruments from a
+    yaml file.
+    Example:
+    >>> scfg = StationConfig('exampleConfig.yaml')
+    >>> dmm1 = scfg.create_instrument('dmm1')
+    """
+
+    PARAMETER_ATTRIBUTES = ['label', 'unit', 'scale', 'inter_delay', 'delay',
+                            'step']
+
+    def __init__(self, filename: str,
+                 station: Optional[Station] = None) -> None:
+        self.monitor_parameters = {}
+
+        if station is None:
+            station = Station.default or Station()
+        self.station = station
+
+        with open(filename, 'r') as f:
+            self.config = yaml.load(f)
+            self._instrument_config = self.config['instruments']
+
+        class ConfigComponent:
+            def __init__(self, data):
+                self.data = data
+
+            def snapshot(self, update=True):
+                return self.data
+
+        # this overwrites any previous station
+        # configurator but does not call the snapshot
+        self.station.components['StationConfigurator'] = ConfigComponent(self.config)
+
+    def load_instrument(self, identifier: str,
+                          **kwargs) -> Instrument:
+        """
+        Creates an instrument driver as described by the loaded config file.
+
+        Args:
+            identifier: the identfying string that is looked up in the yaml
+                configuration file, which identifies the instrument to be added
+            **kwargs: additional keyword arguments that get passed on to the
+                __init__-method of the instrument to be added.
+        """
+
+        # load config
+        if identifier not in self._instrument_config.keys():
+            raise RuntimeError('Instrument {} not found in config.'
+                               .format(identifier))
+        instr_cfg = self._instrument_config[identifier]
+
+        # config is not parsed for errors. On errors qcodes should be able to
+        # to report them
+
+        # check if instrument is already defined and close connection
+        if instr_cfg.get('enable_forced_reconnect',
+                         enable_forced_reconnect):
+            # save close instrument and remove from monitor list
+            with suppress(KeyError):
+                instr = Instrument.find_instrument(identifier)
+                # remove parameters related to this instrument from the monitor list
+                self.monitor_parameters = {k:v for k,v in self.monitor_parameters.items() if v.root_instrument is not instr}
+                instr.close()
+                # remove instrument from station snapshot
+                self.station.components.pop(instr.name)
```

Comment:
If understand this correctly then if `enable_forced_reconnect` is false then the configure instead creates a new instrument in the station.
Is this the behavior we want? I initially assumed the configure would throw an error message and do nothing if `enable_forced_reconnect` is false. Maybe we can add a warning message printed to the console to notify the user that a new instrument instance is created?

Reply:
I think it's going to fail earlier because it's trying to create a new instrument with the same name

Reply:
It's good to see that you already got so much used to the StationConfigurator, that you forgot how it was like before ;-). When you try to create the instrument again, you will get an error from qcodes saying that you can't create an instrument with the same name.

Reply:
the comments crossed...

Reply:
Ahh, I see. Thanks.